Extensions/Guardrail+Settings.swift

```diff
@@ -9,8 +9,76 @@
 import HealthKit
 
 public extension Guardrail where Value == HKQuantity {
-    static let correctionRange = Guardrail(absoluteBounds: 60...180, recommendedBounds: 70...120, unit: .milligramsPerDeciliter)
+    static let suspendThreshold = Guardrail(absoluteBounds: 67...110, recommendedBounds: 74...80, unit: .milligramsPerDeciliter)
 
+    static func maxSuspendThresholdValue(correctionRangeSchedule: GlucoseRangeSchedule?, preMealTargetRange: DoubleRange?, workoutTargetRange: DoubleRange?, unit: HKUnit) -> HKQuantity {
+
+        return [
+            suspendThreshold.absoluteBounds.upperBound.doubleValue(for: unit),
+            correctionRangeSchedule?.minLowerBound().doubleValue(for: unit),
+            preMealTargetRange?.minValue,
+            workoutTargetRange?.minValue
+        ]
+        .compactMap { $0 }
+        .min()
+        .map { HKQuantity(unit: unit, doubleValue: $0) }!
+    }
+
+    static let correctionRange = Guardrail(absoluteBounds: 87...180, recommendedBounds: 101...115, unit: .milligramsPerDeciliter)
+
+    static func minCorrectionRangeValue(suspendThreshold: GlucoseThreshold?, unit: HKUnit) -> HKQuantity {
+        return [
+            correctionRange.absoluteBounds.lowerBound.doubleValue(for: unit),
+            suspendThreshold?.value
```

Comment:
[nit] Consider including `Guardrail.suspendThreshold.absoluteBounds.lowerBound` so that the inherent assumption that the `correctionRange.absoluteBounds.lowerBound` is always greater than the `Guardrail.suspendThreshold.absoluteBounds.lowerBound`. 

(Also in other places in the code that have `suspendThreshold?.value`.)

Reply:
I'm not sure I want to put a check everywhere that ensures any time you read `suspendThreshold` that it has to "clamp" it by `Guardrail.suspendThreshold.absoluteBounds`.  If I start down that road, it means I would have to put a check against `correctionRange.absoluteBounds`, `workoutCorrectionRange.absoluteBounds` and `premealCorrectionRange.absoluteBounds` into the `maxSuspendThresholdValue(...)` function above here.